aiohttp/client.py

```diff
@@ -570,8 +571,11 @@ def request(self,
             # check response status
             if raise_for_status is None:
                 raise_for_status = self._raise_for_status
-            if raise_for_status:
+
+            if raise_for_status is True:
                 resp.raise_for_status()
+            elif raise_for_status is not False:
+                await raise_for_status(resp)  # type: ignore
```

Comment:
It'd be nice if you added a comment explaining why you ignore the type checking and when it's possible to remove it.